crates/slumber_core/src/template/render.rs

```diff
@@ -752,26 +752,65 @@ impl<'a> ChainTemplateSource<'a> {
             self.chain_id.to_string()
         };
 
-        let options = future::try_join_all(options.iter().enumerate().map(
-            |(i, template)| {
-                // Fork the local state, one copy for each new branch
-                let mut stack = stack.clone();
-                async move {
-                    template
-                        .render_chain_config(
-                            format!("options[{i}]"),
-                            context,
-                            &mut stack,
-                        )
-                        .await
+        let resolved_options: Vec<String> = match options {
+            SelectOption::Fixed(options) => {
+                future::try_join_all(options.iter().enumerate().map(
+                    |(i, template)| {
+                        // Fork the local state, one copy for each new branch
+                        let mut stack = stack.clone();
+                        async move {
+                            template
+                                .render_chain_config(
+                                    format!("options[{i}]"),
+                                    context,
+                                    &mut stack,
+                                )
+                                .await
+                        }
+                    },
+                ))
+                .await?
+            }
+            SelectOption::Dynamic(options) => {
+                let mut dynamic_options = options
+                    .render_chain_config("dynamic_options", context, stack)
+                    .await?;
+                // if the chain renders to a non-array value
+                // which is common when this template relies on a prompt
+                // and we're in tui preview, just return that value as
+                // a single option
+                if !dynamic_options.starts_with('[')
+                    && !dynamic_options.ends_with(']')
+                {
+                    dynamic_options = format!("[\"{}\"]", dynamic_options);
                 }
```

Comment:
```suggestion
```
I see the reasoning behind this, but in reality is it going to be common for someone to use a prompt to type in a JSON array, then use that for a dynamic list? I can't think of a real use case for this. This logic introduces other issues, such as:
- If the JSON array has leading or trailing whitespace, it'll get wrapped in another list and you'll end up with just one option in your select, of a stringified list
- Non-list values will silently get turned into single-element lists, rather than failing like they should

Reply:
yea, I hear ya. I struggled with it - felt "sticky" to implement. I'm happy to rip it out and keep it simple. The only case I see this being wonky is during preview for a select chain that builds it's options dynamically based on another prompt 🤷 like you said, probably won't happen in the wild.